### PR TITLE
Timesketch, Turbinia, OSDFIR Infrastructure Initial Helm Charts

### PR DESCRIPTION
**Pull request by wajihyassine**
PR to bootstrap OSDFIR Infrastructure. Closes https://github.com/google/osdfir-infrastructure/issues/2

Includes:
  * Timesketch Helm Chart source code will be hosted under charts/timesketch
  * Turbinia Helm Chart source code will be hosted under charts/turbinia
  * OSDFIR Infrastructure Helm Chart source code will be hosted under charts/osdfir-infrastructure
  * Update Project README with instructions on using the repo/getting started.

### CODE REVIEW

**Comment by wajihyassine** [comment]
Assigning Timesketch and Turbinia leads for review @berggren and @aarontp also FYI @hacktobeer.

Just looking for a second pair of eyes before merging. Most relevant one to review will probably be `values.yaml` file which is what gets generated under the `Parameters` section of each README.

File: charts/timesketch/Chart.lock
Comment (berggren) [inline charts/timesketch/Chart.lock:10]: This will install opensearch 2.x - we don't fully support that yet. Does it work correctly?

File: charts/timesketch/Chart.yaml
Comment (berggren) [inline charts/timesketch/Chart.yaml:25]:
```suggestion
  - name: Johan Berggren
     url: https://github.com/berggren
```

File: charts/timesketch/values-production.yaml
Comment (berggren) [inline charts/timesketch/values-production.yaml:164]: We set `opensearchJavaOpts: "-Xms32g -Xmx32g"` which should be half of the available memory. Should we set this one to 64Gi?

File: charts/timesketch/values-production.yaml
Comment (berggren) [inline charts/timesketch/values-production.yaml:248]: This is a little low for a production deployment - make it something like 500Gi?

File: README.md
Comment (berggren) [inline README.md:13]: What does `my-release` mean here? I opt to make it clear what to use here and what it means. e.g. Is there another release I can deploy?

File: charts/osdfir-infrastructure/README.md
Comment (berggren) [inline charts/osdfir-infrastructure/README.md:100]:
```suggestion
The command removes all the Kubernetes components but Persistent Volumes (PVC) associated with the chart and deletes the release.
```

**Review by berggren — APPROVED** [review]
Over all LGTM for Timesketch. No blockers, but I added a few comments.

File: README.md
Comment (wajihyassine) [reply README.md:13]: Yeah for example you can have a `prod` and `dev` release. I was hoping to note that this can be anything really depending on what type of deployment. If you for example ran helm install prod <timesketch-chart> and helm install dev <timesketch-chart> you will have two timesketch instances running with different release names/pods. I will add a note to make this a little more clear

File: charts/timesketch/Chart.lock
Comment (wajihyassine) [reply charts/timesketch/Chart.lock:10]: I was able to process a timeline and see opensearch index the results. Is there anything I can test for compatibility?

File: charts/timesketch/values-production.yaml
Comment (wajihyassine) [reply charts/timesketch/values-production.yaml:164]: Ah agreed, updated.

File: charts/timesketch/values-production.yaml
Comment (wajihyassine) [reply charts/timesketch/values-production.yaml:248]: Agreed. Updated all prod persistent volumes to 500Gi minimum

File: charts/osdfir-infrastructure/README.md
Comment (aarontp) [inline charts/osdfir-infrastructure/README.md:52]: What does `providing both the original values and the production values` mean exactly?

Comment (aarontp) [reply charts/osdfir-infrastructure/README.md:52]: I think I see now that it means specifying both of those files.  It also looks like this is talking about Turbinia specifically.  Maybe we could clarify that a bit by saying something like `Install the Turbinia chart with the base `values.yaml` config file and the production config file.`

Also, I think this assumes that you are in a specific directory? Should we mention changing to that dir first?

File: charts/osdfir-infrastructure/README.md
Comment (aarontp) [inline charts/osdfir-infrastructure/README.md:138]: FYI, small nit, but it looks like there is a lot of trailing whitespace in this section.

File: charts/osdfir-infrastructure/README.md
Comment (aarontp) [inline charts/osdfir-infrastructure/README.md:244]: Should we disambiguate the different values.yaml (root dir vs. the ones in the turbinia/timesketch directories) and values-production.yaml files?

File: charts/osdfir-infrastructure/README.md
Comment (aarontp) [inline charts/osdfir-infrastructure/README.md:275]: What does this do to the existing PVC, does it extend it in place without deleting anything or does it delete/re-create it?  I'm guessing the former but we should probably add a comment about that  just to be clear.

**Review by aarontp — COMMENTED** [review]
Still going through the config, but here's some quick comments on the doc (mostly just small clarification things).

File: charts/osdfir-infrastructure/Chart.lock
Comment (wajihyassine) [reply charts/osdfir-infrastructure/Chart.lock:3]: As in how to package and push to new releases to the repository? Not yet, but planning to. Would be manual atm until we can setup something to Turbinia/TS with cloud build

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:29]: I'm just trying to note that Turbinia can run on other environments but it won't be able to process cloud based disks except for GCP and any other evidence in case someone for example tries to put it on AWS  and it runs but I guess at that point they will find that out when they try and process a AWS disk there is no option in the CLI to. Should I just point to Turbinia doc for compatibility info?

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:36]: Ah good catch, had missed updating this part

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:44]: Ah no so that should be local to each chart, forgot to add the osdfir-infrastructure one in, but I just added it. I think because of them being subcharts for osdfir then the value mapping wont work the same when referencing them directly

File: README.md
Comment (aarontp) [inline README.md:10]: Won't this command as is fail because of missing required values/fields?  Should we have an example that has the minimum required fields set on the command line here?

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:50]: I was planning to add a link to pull this instead but in meantime updated with the full path

File: charts/timesketch/README.md
Comment (aarontp) [inline charts/timesketch/README.md:39]: What does the `root of the Helm chart` refer to here?  Is that `osdfir-infrastructure` or `osdfir-infrastructure/charts/timesketch`?

File: charts/timesketch/README.md
Comment (aarontp) [inline charts/timesketch/README.md:43]: Should we specify the full path to this file?

File: charts/timesketch/README.md
Comment (aarontp) [inline charts/timesketch/README.md]: Maybe document what dir this should be run from?

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:244]: I think for whatever the user pulls, they should see a single `values.yaml` and `values-production.yaml` per chart. So for example if they used the OSDFIR chart it'd have its own `values.yaml` and `values-production.yaml` (just added) and referencing the Turbinia/Timesketch equivalents wont work. The user also should be pulling from the OCI repo rather than this github repo so when they pull OSDFIR they'd get Turbinia/Timesketch tar'ed up and wouldn't directly see those files

File: charts/timesketch/README.md
Comment (aarontp) [inline charts/timesketch/README.md:253]: Similar comment about documenting resizing vs. deleting and recreating here (as above with Turbinia).

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:52]: It is, but also because Turbinia requires more steps then Timesketch to run on GCP, but the goal was to still install both when running that first install command. I rephrased this to be a bit more clear

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:138]: Ah yeah that was a result of the `readme-generator` not sure if there is a way to remove it, but somehow still formats it nicely when you view the rendered version. Cant look at more configuration options down the line

File: charts/turbinia/Chart.lock
Comment (aarontp) [inline charts/turbinia/Chart.lock:1]: Can you add this (or any of the lock files or things that have hard-coded versions from this PR) to our our release documentation as something that needs to get updated sometimes?

File: charts/turbinia/README.md
Comment (aarontp) [inline charts/turbinia/README.md:18]: This Turbinia link seems to 404 for me?

File: charts/turbinia/README.md
Comment (aarontp) [inline charts/turbinia/README.md:2]: It looks like there is quite a bit of duplicate documentation between the different README pages.  Maybe in a future iteration we can look at how we can reduce that.

File: charts/turbinia/README.md
Comment (aarontp) [inline charts/turbinia/README.md:33]: Should this be with `turbinia-dev` to match above?

File: charts/osdfir-infrastructure/README.md
Comment (wajihyassine) [reply charts/osdfir-infrastructure/README.md:275]: Ack good idea, added a note around how this works but TLDR data would not be deleted/underlying volume is expanded

File: charts/turbinia/README.md
Comment (aarontp) [inline charts/turbinia/README.md:46]: Should we reference the full path here?

File: charts/turbinia/README.md
Comment (aarontp) [inline charts/turbinia/README.md:52]: full path?

File: charts/turbinia/README.md
Comment (aarontp) [inline charts/turbinia/README.md:277]: Similar comment to above  -- which copy of this file is this referring to?

File: charts/turbinia/values-production.yaml
Comment (aarontp) [inline charts/turbinia/values-production.yaml:1]: Maybe add a comment about how the copy in the root dir of this project needs to be updated if this is updated?

**Review by aarontp — COMMENTED** [review]
A few more small comments (mostly similar to previous comments).

File: charts/osdfir-infrastructure/values.yaml
Comment (wajihyassine) [reply charts/osdfir-infrastructure/values.yaml:74]: Fixed

File: charts/osdfir-infrastructure/README.md
Comment (aarontp) [reply charts/osdfir-infrastructure/README.md:29]: Yeah, pointing to Turbinia docs sounds good, thanks.

File: README.md
Comment (wajihyassine) [reply README.md:10]: Per our chat updated this section to be more clear/point to gke installations doc

File: charts/timesketch/README.md
Comment (wajihyassine) [reply charts/timesketch/README.md:39]: It should be either or depending on whether the `osdfir-infrastructure` repo is pulled then it'd be the root of that chart but if it was the `timesketch` chart then the root of that chart from when you run `helm install ../<chart_path>`.

File: charts/timesketch/README.md
Comment (wajihyassine) [reply charts/timesketch/README.md:43]: Per chat, it would be local to wherever the chart was pulled so if timesketch chart was pulled then it'd have its own `values-production.yaml` and osdfir chart will have its own `values-production.yaml`

File: charts/timesketch/README.md
Comment (wajihyassine) [reply charts/timesketch/README.md]: Ack per our chat added a cd instruction right before when the chart is being pulled

File: charts/turbinia/README.md
Comment (wajihyassine) [reply charts/turbinia/README.md:33]: Good catch, fixed

File: charts/turbinia/README.md
Comment (wajihyassine) [reply charts/turbinia/README.md:46]: Per chat not needed anymore

File: charts/turbinia/README.md
Comment (wajihyassine) [reply charts/turbinia/README.md:52]: Per chat not needed anymore or this would still be tools/create-gcp-sa.sh if the Turbinia helm chart was pulled instead

File: charts/turbinia/values-production.yaml
Comment (wajihyassine) [reply charts/turbinia/values-production.yaml:1]: Going to add this as an AI for future maintenance instructions as some more housekeeping items to add

**Comment by JuanPabloSGU** [comment]
Hello @wajihyassine, this looks really promising!!! I have a few things to note for the helm chart that I would recommend looking into when you get the chance. I would recommend that the helm chart be pushed to [Artifact Hub](https://artifacthub.io/) rather then asking the user to `helm pull ... -untar` to allow the use of `value.yaml` files. Another issue is that when using the production step you would also need to add another step in the read me to build the dependencies. You would just need to add a `helm build dependencies` before installing since an error will occur if you are pulling for the first time. Another note that I would recommend is instead of having the release-name to be a static name such as `timesketch-dev` instead have something generic has `<RELEASE-NAME>`. The current docs for the helm chart makes it confusing since we have release names such as timesketch-dev and timesketch-release all in one readme. Again these are recommendations but you have done a lot of work to get to this point and it is greatly appreciated :)

**Comment by aarontp** [comment]
Thanks for the feedback @JuanPabloSGU !  I'll let wajih respond in more details later, but for now I'm going to merge this as is and then we can iterate on it in future PRs.

**Comment by wajihyassine** [comment]
Echoing Aaron, thanks for the feedback @JuanPabloSGU ! In terms of release name, that was a tricky trade off between having something generic then users not understanding the intent of the release name  and using examples to explain it vs having a consistent `<MY-RELEASE>` flag. Will give it some more thought, but overall agreeing with most of the points mentioned and will be addressed in a future PR :)

**Pull request merged**
